addons/web/controllers/main.py

```diff
@@ -702,12 +704,15 @@ def backup(self, master_pwd, name, backup_format = 'zip'):
     @http.route('/web/database/restore', type='http', auth="none", methods=['POST'], csrf=False)
     def restore(self, master_pwd, backup_file, name, copy=False):
         try:
-            data = base64.b64encode(backup_file.read())
-            dispatch_rpc('db', 'restore', [master_pwd, name, data, str2bool(copy)])
+            with tempfile.NamedTemporaryFile(delete=False) as data_file:
```

Comment:
Why not let the contextmanager delete the file?

Reply:
Because I wanted the file closed to avoid potential access and concurrency problems in `db.restore_db()`. Otherwise, I agree this part could be rewritten to take advantage of the context manager.

Reply:
> Because I wanted the file closed to avoid potential access and concurrency problems in `db.restore_db()`. 

Indeed IIRC Windows/NTFS uses exclusive read locks by default, so if the file is still open here it won't be readable by `psql`/`pg_restore`/…

Closing the file should also ensure everything is properly flushed.